src/components/Pressable.tsx

```diff
@@ -0,0 +1,286 @@
+    isPressedDown.current = isPressed;
+    if (typeof props.style === 'function') {
+      setStyleProp(props.style({ pressed: isPressed }));
+    } else {
+      setStyleProp(props.style);
+    }
+    if (typeof props.children === 'function') {
+      setChildrenProp(props.children({ pressed: isPressed }));
+    } else {
+      setChildrenProp(props.children);
+    }
+  };
+
+  const hoverGesture = Gesture.Hover()
+    .onBegin((event) => {
+      setTimeout(
+        () => props.onHoverIn?.(adaptStateChangeEvent(event)),
+        props.delayHoverIn ?? DEFAULT_HOVER_DELAY
+      );
+    })
+    .onEnd((event) => {
+      setTimeout(
+        () => props.onHoverOut?.(adaptStateChangeEvent(event)),
+        props.delayHoverOut ?? DEFAULT_HOVER_DELAY
+      );
+    });
+
+  const touchGesture = Gesture.Manual()
+    .onTouchesDown((event) => {
+      pressableRef.current?.measure((_x, _y, width, height) => {
+        if (
+          !isTouchWithinInset(event.changedTouches.at(-1), normalizedHitSlop, {
+            width,
+            height,
+          }) ||
+          isPressedDown.current
+        ) {
+          return;
+        }
+
+        props.onPressIn?.(adaptTouchEvent(event));
+
+        isPressCallbackEnabled.current = true;
+        setPressedState(true);
+      });
+    })
+    .onTouchesUp((event) => {
+      if (
+        !isPressedDown.current ||
+        event.allTouches.length > event.changedTouches.length
+      ) {
+        return;
+      }
+
+      props.onPressOut?.(adaptTouchEvent(event));
+
+      if (isPressCallbackEnabled.current) {
+        props.onPress?.(adaptTouchEvent(event));
+      }
+
+      setPressedState(false);
+    })
+    .onTouchesCancelled((event) => {
+      if (
+        !isPressedDown.current ||
+        event.allTouches.length > event.changedTouches.length
+      ) {
+        return;
+      }
+
+      // original triggers onPressOut on cancel, but not onPress
+      props.onPressOut?.(adaptTouchEvent(event));
+
+      setPressedState(false);
+    });
+
+  const rippleGesture = Gesture.Native();
+
+  pressGesture.minDuration(props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION);
+  pressGesture.minDuration(props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION);
+
+  const appliedHitSlop = addInsets(
+    normalizedHitSlop,
+    normalizedPressRetentionOffset
+  );
+
+  // this hitSlop block is required by android
+  touchGesture.hitSlop(appliedHitSlop);
+  pressGesture.hitSlop(appliedHitSlop);
+  hoverGesture.hitSlop(appliedHitSlop);
+  rippleGesture.hitSlop(normalizedHitSlop);
+
+  touchGesture.shouldCancelWhenOutside(true);
+  pressGesture.shouldCancelWhenOutside(true);
+  hoverGesture.shouldCancelWhenOutside(true);
+
+  touchGesture.enabled(props.disabled !== true);
+  pressGesture.enabled(props.disabled !== true);
+  hoverGesture.enabled(props.disabled !== true);
+
+  touchGesture.runOnJS(true);
+  pressGesture.runOnJS(true);
+  hoverGesture.runOnJS(true);
```

Comment:
Shouldn't this also be done for `rippleGesture`?

Reply:
Done

Reply:
I don't really see that, did you push the commit?

Reply:
sorry, i took a look and misunderstood it as this line:
> rippleGesture.hitSlop(normalizedHitSlop);

I'm not sure if it's needed to duplicate `enabled` and `shouldCancelWhenOutside`, as it's a blank gesture that doesn't really do anything besides being there for `RNButton`.

I think it'd be better to separate it from these blocks and label it with a comment as to why it is there.
I think it's both cluttering and confusing why this gesture is there to anyone who isn't familiar with nuances of `ripple`.